cpp/src/arrow/compute/kernels/scalar_string_test.cc

```diff
@@ -101,6 +101,11 @@ TEST(TestStringKernels, LARGE_MEMORY_TEST(Utf8Upper32bitGrowth)) {
                                   CallFunction("utf8_upper", {scalar}, options));
 }
 
+TYPED_TEST(TestStringKernels, Utf8Length) {
+  this->CheckUnary("utf8_length", R"(["aaa", null, "áéíóú", "", "b"])",
+                   this->offset_type(), "[3, null, 5, 0, 1]");
```

Comment:
You may add a test with a more complex utf8 encoding, e.g.:
```python
>>> len("ɑɽⱤoW😀")
6
```
(note this is a unicode emoji)

Reply:
Good idea, new tests passed.